inheritance/src/main/java/inheritance/Restaurant.java

```diff
@@ -60,9 +62,18 @@ private void updateStarsRating() {
         }
     }
 
-//Linked List containing the Restaurant's reviews
-    public void addReview(String body, String author, int stars) {
-        insertHeadReview(new Review(body, author, stars));
+    @Override
+    public String getName() {
+        return this.name;
+    }
+
+    //Linked List containing the Restaurant's reviews
+    public void addReview(Review review) {
+        if (listOfReviews.contains(review)) {
+            throw new IllegalArgumentException(String.format("A review for %s has already been submitted.", this.name));
```

Comment:
for(Review review : listOfReviews){
            reviewsJoined.add(review.author + " " + review.body);
        }

Reply:
this is how your code can implement the LinkedList to iterate through everything and build the StringBuilder

Reply:
I tried refactoring this section and it didn't like the change. Does the loop cover not duplicating the review if it is already there?